components/constellation/constellation.rs

```diff
@@ -1020,6 +1023,23 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
+    fn handle_broadcast_storage_event(&self, pipeline_id: PipelineId, storage: StorageType, url: String,
+                                      key: Option<String>, old_value: Option<String>, new_value: Option<String>) {
+        let origin = match Url::parse(&*url) {
+            Ok(url) => url.origin(),
+            Err(err) => return warn!("Failed to parse url {}: {:?}.", url, err),
+        };
+
+        for pipeline in self.pipelines.values() {
+            if (pipeline.id != pipeline_id) && (pipeline.url.origin() == origin) {
```

Comment:
Hmm, I prefer explicit parens rather than relying on operator pecedence to resolve ambiguity.